src/deps.ts

```diff
@@ -111,16 +112,21 @@ type SnappyLib = {
   uncompress(buf: Buffer, opt: { asBuffer: true }): Promise<Buffer>;
 };
 
-export let Snappy: SnappyLib | { kModuleError: MongoMissingDependencyError } = makeErrorModule(
-  new MongoMissingDependencyError(
-    'Optional module `snappy` not found. Please install it to enable snappy compression'
-  )
-);
-
-try {
-  // Ensure you always wrap an optional require in the try block NODE-3199
-  Snappy = require('snappy');
-} catch {} // eslint-disable-line
+export async function getSnappy(): Promise<
```

Comment:
why are we lazy importing?

Reply:
By returning a promise here I'm making calling code handle either `require` or dynamic `import`, so the internal import mechanism we choose is abstracted from the caller.

Reply:
But none of the other imports are async.  Why does snappy need to be async?  We're not using `import`s yet.

Reply:
removed it